internal/util/pool_test.go

```diff
@@ -119,19 +135,44 @@ func TestRunParallelPoolMapCancel(test *testing.T) {
 			cancelFunc()
 		}()
 
-		actual, workErrors, err := RunParallelPoolMap(testCase.numThreads, input, ctx, workFunc)
+		output, err := RunParallelPoolMap(testCase.numThreads, input, ctx, workFunc)
```

Comment:
In not convinced that this test actually tests cancellation cleanly.
All the other tests may finish before the first one actually releases the wait group.

I think a more specific test (no test cases) may be better that uses wait groups to ensure that one piece of work is done, one is never started, and one is in progress when the cancel comes.
Maybe only use one thread.